enhancements/monitoring/alerting-consistency.md

```diff
@@ -0,0 +1,195 @@
+be faced with a disaster.
+
+An example of something that is NOT a critical alert:
+* MCO and/or related components are completely dead/crash looping.
+
+Even though the above is quite significant, there is no risk of immediate loss
+of the cluster.
+
+You might be thinking to yourself: "But my component is **super** important, and
+if it's not working, the user can't do X, which user definitely will want."  All
+that is probably true, but that doesn't make your alert "critical" it just makes
+it worthy of an alert.
+
+### Warning Alerts
+
+TL/DR: fix this soon, or some things won't work and upgrades will probably be
+blocked.
+
+If your alert does not meet the criteria in "Critical Alerts" above, it belongs
+to the warning level or lower.
+
+Most alerts probably fit in this category.  This level should represent degraded
+or unavailable services that do not jeopardize the immediate life of cluster.
+
+Timeline:  ~60 minutes
+
+60 minutes?  That seems high!  That's because it is high.  We want to reduce
+the noise.  We've done a lot to make clusters and operators auto-heal
+themselves.  The whole idea is that if a condition has persisted for more than
+60 minutes, it's unlikely to be resolved without intervention any time later.
+
+#### Example 1: All machine-config-daemons are crashlooping.
+
+That seems significant
+and it probably is, but nobody has to get out of bed to fix this **right now**.
+But, what if a machine dies, I won't be able to get a replacement?  Yeah, that
+is probably true, but how often does that happen?  Also, that's an entirely
+separate set of concerns.
+
+#### Example 2:  I have an operator that needs at least 3 of something to work.
+* With 2/3 replicas, warning alert after 60M.
+* With 1/3 replicas, warning alert after 10M.
+* With 0/3 replicas, warning alert after 5M.
+
+Q: How long should we wait until the above conditions rises to a 'critical'
+alert?
+
+A: It should never rise to the level of a critical alert, it's not critical.  If
+it was, this section would not apply.
+
+#### Example 3:  I have an operator that only needs 1 replica to function
+
+Well, if the cluster can upgrade with only 1 replica, and the the service is
+not unavailable despite other replicas being unavailable, this can probably
+be just an info-level alert.  We should have a "minor" alert, TBD.  But for now,
+"info" is the closest thing.
+
+#### Q: What if a particular condition will block upgrades?
+
+A: It's a warning level alert.
+
+
+### Documentation Required
+
+Every alert needs to have a URL to relevant documentation to resolve the issue.
+
+If there's not documentation, there shouldn't be an alert.
+
+Sometimes an alert's impact is not obvious.  We should state the impact of the
+alert.  Don't expect users to understand the context, we have too many moving
+pieces.
+
+
+### Open Questions [optional]
+
+Frankly, having just "Critical" and "Warning" is not helpful.
```

Comment:
`Info` exists and is encouraged to be used

Reply:
Personally, I think "Info" is not something that should be in 'alerts.'  We need a different channel for informational messaging.

Reply:
The community isn't fully decided on what is the way to go, however with us inheriting a major set of OCP alerts from https://github.com/kubernetes-monitoring/kubernetes-mixin it makes sense to obey the [alert severities defined there](https://github.com/kubernetes-monitoring/kubernetes-mixin#alert-severities)

Reply:
Which alerts are we inheriting?

Reply:
All of them 
https://github.com/openshift/cluster-monitoring-operator/blob/721bc047264fb4f52581f9c595fd31a304c6eaf0/jsonnet/jsonnetfile.json#L16

https://github.com/prometheus-operator/kube-prometheus/blob/de2d1b523135cd4e561b899b294f015e52e4f704/jsonnet/kube-prometheus/jsonnetfile.json#L43